CMakeLists.txt

```diff
@@ -263,215 +235,52 @@ if(MSVC AND NOT DEFINED CXX_FLAGS_MSVC)
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Winit-self"
-	HAS_COMPILER_FLAG_WINIT_SELF
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wlogical-op"
-	HAS_COMPILER_FLAG_WLOGICAL_OP
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wmissing-declarations"
-	HAS_COMPILER_FLAG_WMISSING_DECLARATIONS
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wredundant-decls"
-	HAS_COMPILER_FLAG_WREDUNDANT_DECLS
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wctor-dtor-privacy"
-	HAS_COMPILER_FLAG_WCTOR_DTOR_PRIVACY
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wnon-virtual-dtor"
-	HAS_COMPILER_FLAG_WNON_VIRTUAL_DTOR
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wdouble-promotion"
-	HAS_COMPILER_FLAG_WDOUBLE_PROMOTION
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wmismatched-tags"
-	HAS_COMPILER_FLAG_WMISMATCHED_TAGS
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wuseless-cast"
-	HAS_COMPILER_FLAG_WUSELESS_CAST
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wc++11-compat"
-	HAS_COMPILER_FLAG_WCXX_11_COMPAT
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wnoexcept"
-	HAS_COMPILER_FLAG_WNOEXCEPT
-)
-
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wdocumentation"
-	HAS_COMPILER_FLAG_WDOCUMENTATION
-)
+if(ENABLE_PEDANTIC_COMPILATION)
 
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wdocumentation-deprecated-sync"
-	HAS_COMPILER_FLAG_WDOCUMENTATION
-	"-Wno-documentation-deprecated-sync"
-)
+	set(CXX_FLAGS_PEDANTIC_COMPILATION "-Wlogical-op -Wmissing-declarations -Wredundant-decls -Wctor-dtor-privacy -Wnon-virtual-dtor -Wdouble-promotion -Wuseless-cast -Wnoexcept")
 
-check_compiler_has_flag(
-	CXX_FLAGS_PEDANTIC_COMPILATION
-	"-Wextra-semi"
-	HAS_COMPILER_FLAG_WEXTRA_SEMI
-)
+	if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+		set(CXX_FLAGS_PEDANTIC_COMPILATION "${CXX_FLAGS_PEDANTIC_COMPILATION} -Wmismatched-tags -Wdocumentation -Wno-documentation-deprecated-sync")
+	endif()
+	
+	set(COMPILER_FLAGS "${COMPILER_FLAGS} ${CXX_FLAGS_PEDANTIC_COMPILATION}")
 
-check_compiler_has_flag(
-	CXX_FLAGS_STRICT_COMPILATION
-	"-Wconditional-uninitialized"
-	HAS_COMPILER_FLAG_WCONDITIONAL_INITIALIZED
-)
+endif(ENABLE_PEDANTIC_COMPILATION)
 
 ### Set the final compiler flags.
 
-set(COMPILER_FLAGS "${CXX_FLAGS_PROJECT}")
 if(MSVC)
 	set(COMPILER_FLAGS "${CXX_FLAGS_MSVC} ${COMPILER_FLAGS}")
 endif(MSVC)
-if(ENABLE_STRICT_COMPILATION)
-	set(COMPILER_FLAGS "${COMPILER_FLAGS} ${CXX_FLAGS_STRICT_COMPILATION}")
-endif(ENABLE_STRICT_COMPILATION)
-if(ENABLE_PEDANTIC_COMPILATION)
-	set(COMPILER_FLAGS "${COMPILER_FLAGS} ${CXX_FLAGS_PEDANTIC_COMPILATION}")
-endif(ENABLE_PEDANTIC_COMPILATION)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+	set(COMPILER_FLAGS "${COMPILER_FLAGS} -Qunused-arguments -Wno-unknown-warning-option -Werror=non-virtual-dtor")
```

Comment:
Doesn't look very logical to me that `-Werror=non-virtual-dtor` is enabled by default with Clang but requires pedantic compilation mode with GCC.

I suggest enabling the error with both compilers by default.

Reply:
Ah, whoops - I copied those three warnings straight from [scons+clang](https://github.com/wesnoth/wesnoth/blob/master/SConstruct#L566), and didn't notice that it was duplicated.